### PR TITLE
Simulation workflow implementation

### PR DESCRIPTION
**Pull request by zeroset**
This PR is a draft of the simulation workflow.

### CODE REVIEW

**Comment by JetteSchumann** [comment]
Notes on discussion with @schroedtert:
- merge #37 into #57 to create one PR for the basic simulation workflow
- we should discuss what is done in the simulation loop on python and c++ side. Maybe: python - loop over time, C++ - loop over agents
- simulation context must contain information about: agents, world, model parameters and model results
- next steps:
  - World Implementation and integration in simulation flow @JetteSchumann 
  - read in/create Agents and integrate in simulation flow @schroedtert

File: cpp/pycore/source/simulation_binding.cpp [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 #include <simulation.hpp>
 
 void bind_simulation(pybind11::module_ & m)
 {
     pybind11::class_<jps::Simulation>(m, "Simulation")
         .def(pybind11::init<>())
-        .def("get", &jps::Simulation::getValue);
+        .def("get", &jps::Simulation::getValue)
+        //.def("steps", [](jps::Simulation const & simulation) {
+        //} )
```
Comment (schroedtert) [inline cpp/pycore/source/simulation_binding.cpp:12]: Please remove commented code and open Issue for the missing pybinds

File: jps/jupedsim/util/simulation_files.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import pathlib
+
+
+class SimulationFiles:
+    def __init__(self, path: pathlib.Path) -> None:
+        self.world_dxf = path / "world.dxf"
+        if not self.world_dxf.exists():
+            raise FileNotFoundError(
+                errno.ENOENT, os.strerror(errno.ENOENT), self.world_dxf
+            )
```
Comment (schroedtert) [inline jps/jupedsim/util/simulation_files.py:12]: Do we need already test for this? And we should document somewhere, that one needs a `world.dxf` to start a simulation.

**Review by schroedtert — CHANGES_REQUESTED** [review]

File: jps/jupedsim/util/simulation_files.py (hunk 2 above)
Comment (zeroset) [reply jps/jupedsim/util/simulation_files.py:12]: Thats just for the beginning. We can change the file name retrieval later.... I add this to a general Documentation Issue. #63

**Review by schroedtert — APPROVED** [review]

**Pull request merged**